backend/tracim_backend/applications/share/lib.py

```diff
@@ -0,0 +1,172 @@
+                share_group_id=share_group_id,
+                enabled=True,
+            )
+            self.save(content_share)
+            content_shares.append(content_share)
+            self._session.flush()
+
+        if do_notify and self._config.EMAIL__NOTIFICATION__ACTIVATED:
+            try:
+                email_manager = self._get_email_manager(self._config, self._session)
+                email_manager.notify__share__content(
+                    emitter=self._user,
+                    shared_content=content,
+                    content_share_receivers=self.get_content_shares_in_context(content_shares),
+                    share_password=password,
+                )
+            # FIXME - G.M - 2018-11-02 - hack: accept bad recipient user creation
+            # this should be fixed to find a solution to allow "fake" email but
+            # also have clear error case for valid mail.
+            except SMTPRecipientsRefused:
+                logger.warning(
+                    self,
+                    "Share initied by {email} but SMTP server refuse to send notification".format(
+                        email=self._user.email
+                    ),
+                )
+            except SMTPException as exc:
+                raise NotificationSendingFailed(
+                    "Notification for shared file can't be send " "(SMTP error)."
+                ) from exc
+        return content_shares
+
+    def _get_email_manager(self, config: CFG, session: Session):
+        """
+        :return: EmailManager instance
+        """
+        smtp_config = SmtpConfiguration(
+            config.EMAIL__NOTIFICATION__SMTP__SERVER,
+            config.EMAIL__NOTIFICATION__SMTP__PORT,
+            config.EMAIL__NOTIFICATION__SMTP__USER,
+            config.EMAIL__NOTIFICATION__SMTP__PASSWORD,
+        )
+
+        return ShareEmailManager(config=config, smtp_config=smtp_config, session=session)
+
+    def get_content_shares(self, content: Content) -> typing.List[ContentShare]:
+        return self.base_query().filter(ContentShare.content_id == content.content_id).all()
+
+    def get_content_share_in_context(self, content_share: ContentShare) -> ContentShareInContext:
+        return ContentShareInContext(content_share, self._session, self._config, self._user)
+
+    def get_content_shares_in_context(
+        self, content_shares: typing.List[ContentShare]
+    ) -> typing.List[ContentShareInContext]:
+        content_shares_in_context = []
+        for content_share in content_shares:
+            content_shares_in_context.append(self.get_content_share_in_context(content_share))
+        return content_shares_in_context
+
+    def get_content_share_by_token(self, share_token: str):
+        return (
+            self._session.query(ContentShare).filter(ContentShare.share_token == share_token).one()
+        )
+
+    def check_password(self, content_share: ContentShare, password: typing.Optional[str]) -> None:
```

Comment:
Add docstring with which exception is raised

Reply:
done.